dev-server/ui-playground/index.js

```diff
@@ -3,11 +3,15 @@ import ButtonPatterns from './components/ButtonPatterns';
 
 import sidebarIcons from '../../src/sidebar/icons';
 
+/** @type {import('@hypothesis/frontend-shared/lib/pattern-library').PlaygroundRoute['group']} */
```

Comment:
It'll be easier to add the cast to the `extraRoute` definition:

```js
/** @type {import('@hypothesis/frontend-shared/lib/pattern-library').PlaygroundRoute[]} */
const extraRoutes = [
  {
    route: '/buttons',
    title: 'Buttons',
    component: ButtonPatterns,
    group: 'components',
  },
];
 ```

This way you only have to specify a type once if there are multiple fields like `group`. You'll also get an earlier error if the `extraRoutes` object has the wrong structure.

Reply:
Much better, thanks.

An alternative approach that will be available with typescript 4.5 is:

` group: /** @type {const} */ ('components')`

https://github.com/microsoft/TypeScript/issues/30445

Reply:
Ah, that's useful to know. I think I'd still prefer to type `extraRoutes` in this particular case in order to communicate intent to readers.